test/funnels/Depositor.t.sol

```diff
@@ -0,0 +1,566 @@
+    bytes constant DAI_USDC_PATH = abi.encodePacked(DAI, uint24(100), USDC);
+
+    address constant DAI           = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address constant USDC          = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address constant DAI_USDC_POOL = 0x5777d92f208679DB4b9778590Fa3CAB3aC9e2168;
+    address constant UNIV3_ROUTER  = 0xE592427A0AEce92De3Edee1F18E0157C05861564;
+    address constant UNIV3_FACTORY = 0x1F98431c8aD98523631AE4a59f267346ea31F984;
+
+    address constant FACILITATOR    = address(0x1337);
+    uint8   constant DEPOSITOR_ROLE = uint8(2);
+
+    int24 constant REF_TICK = -276324; // tick corresponding to 1 DAI = 1 USDC calculated as ~= math.log(10**(-12))/math.log(1.0001)
+
+    function setUp() public {
+        vm.createSelectFork(vm.envString("ETH_RPC_URL"));
+        
+        buffer = new AllocatorBuffer();
+        roles = new AllocatorRoles();
+        depositor = new Depositor(address(roles), ilk, UNIV3_FACTORY, address(buffer));
+
+        roles.setIlkAdmin(ilk, address(this));
+        roles.setRoleAction(ilk, DEPOSITOR_ROLE, address(depositor), depositor.deposit.selector, true);
+        roles.setRoleAction(ilk, DEPOSITOR_ROLE, address(depositor), depositor.withdraw.selector, true);
+        roles.setRoleAction(ilk, DEPOSITOR_ROLE, address(depositor), depositor.collect.selector, true);
+        roles.setUserRole(ilk, FACILITATOR, DEPOSITOR_ROLE, true);
+
+        depositor.setLimits(DAI, USDC, 3600 seconds, uint128(10_000 * WAD), uint128(10_000 * 10**6));
+
+        deal(DAI,  address(buffer), 1_000_000 * WAD,   true);
+        deal(USDC, address(buffer), 1_000_000 * 10**6, true);
+        buffer.approve(USDC, address(depositor), type(uint256).max);
+        buffer.approve(DAI,  address(depositor), type(uint256).max);
+    }
+
+    function testConstructor() public {
+        Depositor d = new Depositor(address(0xBEEF), "SubDAO 1", address(0xAAA), address(0xCCC));
+        assertEq(address(d.roles()),  address(0xBEEF));
+        assertEq(d.ilk(), "SubDAO 1");
+        assertEq(d.uniV3Factory(), address(0xAAA));
+        assertEq(d.buffer(), address(0xCCC));
+        assertEq(d.wards(address(this)), 1);
+    }
+
+    function testAuth() public {
+        checkAuth(address(depositor), "Depositor");
+    }
+
+    function testModifiers() public {
+        bytes4[] memory authedMethods = new bytes4[](3);
+        authedMethods[0] = depositor.deposit.selector;
+        authedMethods[1] = depositor.withdraw.selector;
+        authedMethods[2] = depositor.collect.selector;
+
+        vm.startPrank(address(0xBEEF));
+        checkModifierForLargeArgs(address(depositor), "Depositor/not-authorized", authedMethods);
+        vm.stopPrank();
+    }
+
+    function testSetLimits() public {
+        vm.expectEmit(true, true, true, true);
+        emit SetLimits(address(1), address(2), 3, 4, 5);
+        vm.prank(address(this)); depositor.setLimits(address(1), address(2), 3, 4, 5);
```

Comment:
Same here, we should probably test with some `zzz` value different than 0.

Reply:
Done.